app/src/main/java/org/simple/clinic/enterotp/EnterOtpModel.kt

```diff
@@ -15,24 +15,31 @@ data class EnterOtpModel(
     val isAsyncOperationOngoing: Boolean,
     val protectedState: ProtectedState,
 
+    val minOtpRetries: Int
     ) : Parcelable {
 
   companion object {
 
-    fun create(): EnterOtpModel {
+    fun create(
+        minOtpRetries: Int
+    ): EnterOtpModel {
       return EnterOtpModel(
           user = null,
           otpValidationResult = ValidationResult.NotValidated,
           asyncOpError = null,
           isAsyncOperationOngoing = false,
-          protectedState = ProtectedState.Allowed(0,5)
+          protectedState = ProtectedState.Allowed(0,5),
```

Comment:
Hmm, I also see we are hardcoding 5 here for the `ProtectedState.Allowed` constructor. If that is meant to be the same as a limit of failed attempts configuration property, then this should be accepted via a dependency instead of being hardcoded as well.